src/plat/sun/detect.c

```diff
@@ -284,7 +284,10 @@ void detect_res(void)
 		safe_strncpy(res_str, "No X Server", MAX_STRLEN);
 
 		if (error)
+		{
 			ERR_REPORT("Problem detecting X display resolution.");
+			return;
```

Comment:
Instead of returning here, let's just move the `XCloseDisplay` below into the `if` block.

Reply:
(See `detect_res` in `plat/linux/detect.c`)

Reply:
Amended.

Reply:
Feel free to toss the pull request and make the change how you like if it's not right and that's easier.

Reply:
> Feel free to toss the pull request and make the change how you like if it's not right and that's easier.

It looks good to me! I'll merge it in a bit.